tests/test_integration.py

```diff
@@ -55,90 +77,100 @@ def pg_install_tables(container):
+    paginator = client.get_paginator("list_objects")
+    for result in paginator.paginate(Bucket="ooni-data", Delimiter="/", Prefix=prefix):
+        for f in result.get("Contents", []):
+            fkey = f.get("Key")
+            dst_pathname = os.path.join(dst_bucket_dir, os.path.basename(fkey))
             try:
                 s = os.stat(dst_pathname)
-                if s.st_size == f.get('Size'):
-                    print('SKIP')
+                if s.st_size == f.get("Size"):
                     continue
-            except Exception: # XXX maybe make this more strict. It's FileNotFoundError on py3 and OSError on py2
+            except Exception:  # XXX maybe make this more strict. It's FileNotFoundError on py3 and OSError on py2
                 pass
-            resource.meta.client.download_file('ooni-data', fkey, dst_pathname)
+            print("[+] Downloading {}".format(dst_pathname))
+            resource.meta.client.download_file("ooni-data", fkey, dst_pathname)
 
-def run_centrifugation(client, bucket_date):
-    fmt = '%Y-%m-%d'
+
+def run_centrifugation(docker_client, bucket_date):
+    ## Run centrifugation in a dedicated container based on
+    ## the openobservatory/pipeline-shovel image
+    fmt = "%Y-%m-%d"
     end_bucket_date = (datetime.strptime(bucket_date, fmt) + timedelta(1)).strftime(fmt)
 
-    centrifugation_cmd = '/mnt/af/shovel/centrifugation.py --start {}T00:00:00'.format(bucket_date)
-    centrifugation_cmd += ' --end {}T00:00:00'.format(end_bucket_date)
-    centrifugation_cmd += " --autoclaved-root /mnt/testdata --postgres 'host={} user={}'".format(METADB_NAME, METADB_PG_USER)
+    centrifugation_cmd = "/mnt/af/shovel/centrifugation.py --start {}T00:00:00".format(
+        bucket_date
+    )
+    centrifugation_cmd += " --end {}T00:00:00".format(end_bucket_date)
+    centrifugation_cmd += " --autoclaved-root /mnt/testdata --postgres 'host={} user={}'".format(
+        METADB_NAME, METADB_PG_USER
+    )
 
     start_time = time.time()
 
     print("running shovel @{}: {}".format(start_time, centrifugation_cmd))
-    shovel_container = client.containers.run(
-        'openobservatory/pipeline-shovel:latest',
+    image_name = "openobservatory/pipeline-shovel:" + conf.shovel_image_tag
+    print("using image " + image_name)
+    shovel_container = docker_client.containers.run(
+        image_name,
         centrifugation_cmd,
-        mem_limit='512m',
-        user='{}:{}'.format(os.getuid(),os.getgid()),
-        working_dir='/mnt',
-        links={
-            METADB_NAME: None
-        },
-        volumes={
-            REPO_ROOT: {
-                'bind': '/mnt',
-                'mode': 'rw'
-            }
-        },
+        mem_limit="512m",
+        user="{}:{}".format(os.getuid(), os.getgid()),
+        working_dir="/mnt",
+        links={METADB_NAME: None},
+        volumes={REPO_ROOT: {"bind": "/mnt", "mode": "rw"}},
         privileged=True,
```

Comment:
Do you know why this container needs to be `--privileged`?

Reply:
Unsure. It seems to work without it.

Reply:
I suggest to mention this in a comment as a possible future improvement?

Reply:
@hellais can we also take note of this?

Reply:
I filed an issue for it here: https://github.com/ooni/pipeline/issues/211.